packages/test/test-end-to-end-tests/src/test/gc/gcSweepAttachmentBlobs.spec.ts

```diff
@@ -0,0 +1,1063 @@
+
+				// Add the blob handles to reference them.
+				mainDataStore._root.set("blob1", blobHandle1);
+				mainDataStore._root.set("blob2", blobHandle2);
+
+				// Upload the same blob. This will get de-duped and we will get back another handle.
+				const blobHandle3 = await mainDataStore._context.uploadBlob(
+					stringToBuffer(blobContents, "utf-8"),
+				);
+				assert.notStrictEqual(
+					blobHandle1.absolutePath,
+					blobHandle3.absolutePath,
+					"blob handles should be different",
+				);
+
+				// Add the new handle and then remove all the handles to unreference the blob.
+				mainDataStore._root.set("blob3", blobHandle2);
+				mainDataStore._root.delete("blob1");
+				mainDataStore._root.delete("blob2");
+				mainDataStore._root.delete("blob3");
+
+				// Summarize so that the above attachment blobs are marked unreferenced.
+				await provider.ensureSynchronized();
+				await summarizeNow(summarizer);
+
+				// Wait for sweep timeout so that the blob is ready to be deleted.
+				await delay(sweepTimeoutMs + 10);
+
+				// Send an op to update the current reference timestamp that GC uses to make sweep ready objects.
+				mainDataStore._root.set("key", "value");
+				await provider.ensureSynchronized();
+
+				// Summarize so that the blobs are deleted.
+				const summary2 = await summarizeNow(summarizer);
+
+				// Load a new container from the above summary which should note have the blobs.
+				const container2 = await loadContainer(summary2.summaryVersion);
+
+				// Retrieving the blob via any of the handles should fail. Note that the blob is requested via its url since
+				// this container does not have access to the blob's handle.
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle1.absolutePath,
+					"Container2: Blob1",
+				);
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle2.absolutePath,
+					"Container2: Blob2",
+				);
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle3.absolutePath,
+					"Container2: Blob3",
+				);
+
+				// Retrieving the blobs in the summarizer container should fail as well.
+				await validateBlobRetrievalFails(
+					summarizerContainer,
+					blobHandle1.absolutePath,
+					"Summarizer: Blob1",
+				);
+			},
+		);
+	});
+
+	describe("Deleted blob in summary", () => {
+		/**
+		 * Validates that the given blob state is correct in the summary:
+		 * - It should be no blob tree in the summary.
+		 * - The blob should not be present in the GC state in GC summary tree.
+		 * - The blob should be present in the deleted nodes in GC summary tree.
+		 */
+		function validateBlobStateInSummary(summaryTree: ISummaryTree, blobNodePath: string) {
+			// Validate that the blob tree should not be in the summary since there should be no attachment blobs.
+			const blobsTree = summaryTree.tree[blobsTreeName] as ISummaryTree;
+			assert(blobsTree === undefined, "Blobs tree should be present in the summary");
```

Comment:
Shouldn't this assert be defined? Or should the text of the assert change, or am I missing something?

Reply:
The assert text should be changed - https://github.com/microsoft/FluidFramework/pull/14249